### PR TITLE
[stable/cockroachdb] Used storageClassName for cockroachdb.

### PR DESCRIPTION
**Pull request by gyliu513**
Fixed cockroachdb for #1869

/cc @unguiculus

### CODE REVIEW

**Comment by prydonius** [comment]
@gyliu513 this seems incomplete, can we follow the same format in your other PRs?

**Comment by gyliu513** [comment]
@prydonius done

File: stable/cockroachdb/templates/cockroachdb-statefulset.yaml [hunk 1]
```diff
@@ -173,11 +173,16 @@ spec:
...
-        volume.alpha.kubernetes.io/storage-class: "{{ .Values.StorageClass }}"
     spec:
       accessModes:
         - "ReadWriteOnce"
       resources:
         requests:
           storage: "{{ .Values.Storage }}"
+{{- if .Values.StorageClass }}
+{{- if (eq "-" .Values.StorageClass) }}
+  storageClassName: ""
```
Comment (prydonius) [inline stable/cockroachdb/templates/cockroachdb-statefulset.yaml]: This is at the wrong indent level, it should be under `spec`

**Review by prydonius — CHANGES_REQUESTED** [review]

File: stable/cockroachdb/templates/cockroachdb-statefulset.yaml (hunk 1 above)
Comment (gyliu513) [reply stable/cockroachdb/templates/cockroachdb-statefulset.yaml]: My bad, thanks @prydonius

File: stable/cockroachdb/values.yaml [hunk 2]
```diff
@@ -19,6 +19,13 @@ Resources:
...
     memory: "512Mi"
 Storage: "1Gi"
+## cockroachdb data Persistent Volume Storage Class
+## If defined, StorageClass: <storageClass>
+## If set to "-", StorageClass: "", which disables dynamic provisioning
+## If undefined (the default) or set to null, no storageClassName spec is
+##   set, choosing the default provisioner.  (gp2 on AWS, standard on
+##   GKE, AWS & OpenStack)
+##
 StorageClass: "anything"
```
Comment (prydonius) [inline stable/cockroachdb/values.yaml]: This should be commented to:
```
StorageClass: "-"
```
to be consistent with the others

Comment (gyliu513) [reply stable/cockroachdb/values.yaml]: Done

**Comment by unguiculus** [comment]
This is now obsolete.

**Pull request closed without merging**